api/v1beta1/coordinator_selection.go

```diff
@@ -0,0 +1,56 @@
+/*
+ * coordinator_selection.go
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1beta1
+
+import "math"
+
+// CoordinatorSelectionSetting defines the process class and the priority of it.
+// A higher priority means that the process class is preferred over another.
+type CoordinatorSelectionSetting struct {
+	ProcessClass ProcessClass `json:"processClass,omitempty"`
+	Priority     int          `json:"priority,omitempty"`
+}
+
+// IsEligibleAsCandidate checks if the given process has the right process class to be considered a valid coordinator
+func (cluster *FoundationDBCluster) IsEligibleAsCandidate(pClass ProcessClass) bool {
+	if len(cluster.Spec.CoordinatorSelection) == 0 {
+		return pClass.IsStateful()
```

Comment:
The way this check is structured suggests that if someone expressly adds a stateless class to the CoordinatorSelection list, it would allow recruiting that as a coordinator. I think that would open the door to dangerous behavior, since we wouldn't be giving that process a persistent volume.

Reply:
That was also a concern but I was not sure if we want to limit the coordinators to only stateful services but I guess that makes sense.